services/workflows-service/src/note/note.repository.ts

```diff
@@ -0,0 +1,102 @@
+import { PrismaService } from '@/prisma/prisma.service';
+import { Note, Prisma, PrismaClient } from '@prisma/client';
+import { Injectable } from '@nestjs/common';
+import { PrismaTransaction } from '@/types';
+
+const defaultFieldsSelect = {
+  id: true,
+  entityId: true,
+  entityType: true,
+  noteableId: true,
+  noteableType: true,
+  content: true,
+  fileIds: true,
+  createdAt: true,
+  createdBy: true,
+  updatedAt: true,
+};
+
+const defaultArgs = {
+  select: {
+    ...defaultFieldsSelect,
+    parentNote: { select: defaultFieldsSelect },
+    childrenNotes: { select: defaultFieldsSelect },
+  },
+  orderBy: {
+    createdAt: 'desc',
+  },
+};
+
+@Injectable()
+export class NoteRepository {
+  constructor(protected readonly prismaService: PrismaService) {}
+
+  async create(
+    data: Omit<Prisma.NoteUncheckedCreateInput, 'projectId'>,
+    projectId: string,
```

Comment:
Isn't projectId required on the create input anyways?
```suggestion
    data: Prisma.NoteCreateInput,
```

Reply:
It is, it just doesn't come inside `data`, but as a separate property after it (which is manually add when I actually create the note)